src/docfx/lib/git/GitRepoInfo.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Microsoft.Docs.Build
+{
+    internal class GitRepoInfo
+    {
+        private static readonly Regex GitHubRepoUrlRegex =
+            new Regex(
+                @"^((https|http):\/\/(.+@)?github\.com\/|git@github\.com:)(?<account>\S+)\/(?<repository>[A-Za-z0-9_.-]+)(\.git)?\/?$",
+                RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.RightToLeft);
+
+        public GitHost Host { get; set; }
+
+        public string Account { get; set; }
+
+        public string Name { get; set; }
+
+        public string Branch { get; set; }
+
+        public string HeadCommitId { get; set; }
+
+        public string RootPath { get; set; }
```

Comment:
>RootPath [](start = 22, length = 8)

nit: RepositoryPath?

Reply:
class name already contains "Repo"

---
In reply to: [195016528](https://github.com/dotnet/docfx/pull/2876#discussion_r195016528) [](ancestors = 195016528)

Reply:
There are too many path here, so we say it explicitly. E.g., in Docset, there is a DocsetPath property.

---
In reply to: [195022098](https://github.com/dotnet/docfx/pull/2876#discussion_r195022098) [](ancestors = 195022098,195016528)